scripts/edit-task.js

```diff
@@ -0,0 +1,231 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+import createButton from "/scripts/factory/createButton.js";
+import toggleButtons from "/scripts/factory/toggleButton.js";
+
+function saveEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  tasks[id].content = inputElement.textContent;
+  toggleButtons(
+    editButton,
+    deleteButton,
+    revertButton,
+    saveButton,
+    doneButton,
+    saveAndDoneButton
+  );
+  console.log("call");
+  saveButton.removeEventListener(CLICK_EVENT, saveEventHandler);
+  revertButton.removeEventListener(CLICK_EVENT, revertEventHandler);
+  saveAndDoneButton.removeEventListener(CLICK_EVENT, saveAndDoneEventHandler);
+}
+
+function onTodoDone(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  toolbar,
+  tasks,
+  id,
+  editEventHandler,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  inputElement.classList.add("done");
+  const task = tasks[id];
+  task.done = true;
+  task.content = inputElement.textContent;
+
+  editButton.removeEventListener(CLICK_EVENT, editEventHandler);
+  // doneButton.removeEventListener(CLICK_EVENT, onTodoDone);
```

Comment:
dead/commented out code should not be committed.

```suggestion
  editButton.removeEventListener(CLICK_EVENT, editEventHandler);
```